dace/data.py

```diff
@@ -133,7 +133,7 @@ def __init__(self,
                  transient=False,
                  storage=dace.dtypes.StorageType.Default,
                  allow_conflicts=False,
-                 location='',
+                 location={},
```

Comment:
That's generally not good in Python. It'll create one mutable object that if someone changes, will change for everyone who call this function. Please use `=None`  instead and `self.location = location or {}` inside the method.